metadata-ingestion/tests/integration/db2/test_db2.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+import re
+import subprocess
+
+import pytest
+import sqlalchemy
+import yaml
+
+from datahub.ingestion.run.pipeline import Pipeline
+from datahub.testing import mce_helpers
+from tests.test_helpers.docker_helpers import wait_for_port
+
+logger = logging.getLogger(__name__)
+
+DB2_PORT = 50000
+
+
+@pytest.fixture(scope="module")
+def test_resources_dir(pytestconfig):
+    return pytestconfig.rootpath / "tests/integration/db2"
+
+
+def is_db2_up(container_name: str) -> bool:
+    cmd = f"docker logs {container_name} 2>&1 | grep 'Setup has completed.'"
+    ret = subprocess.run(
+        cmd,
+        shell=True,
+    )
+    return ret.returncode == 0
+
+
+def _split_statements(sql):
+    """Split a SQL script into individual statements that can be executed.
+    Statements are usually separated by semicolons, except that BEGIN/END
+    blocks are kept as a single item."""
+    statements: list[str] = []
+    inside_begin = False
+    sql = re.sub(r"--[^\n]*", "", sql)
+    for statement in sql.split(";"):
+        statement = statement.strip()
+        logger.warning(statement)
+        if not statement:
+            continue
+        if inside_begin and statements:
+            statements[-1] += "; " + statement
+        else:
+            statements.append(statement)
+        if "BEGIN" in statement.upper():
+            inside_begin = True
+        if statement.upper() == "END":
+            inside_begin = False
+    return statements
+
+
+@pytest.fixture(scope="module")
+def db2_runner(docker_compose_runner, pytestconfig, test_resources_dir):
+    with docker_compose_runner(
```

Comment:
@sgomezvillamor it would be nice to be able to skip Docker Compose and repoint this at an existing database service, for faster local testing and i/AS400 testing. Do we have any existing patterns for that? Maybe an environment variable?